### PR TITLE
refactor(bom): add Identity repository

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds the Identity repository to the BOM, ensuring Identity artifacts are always found properly. This also fixes the release process which is currently failing to build due to Identity not being available.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
Here's the successful test run: https://github.com/camunda/zeebe/actions/runs/6640129975

I had to add an additional commit which ensures the workflow uses this branch for the release, which I will revert once this is accepted before merging.

File: .github/workflows/release-main-dry-run.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           - version: 0.8.2-alpha1
             latest: false
     with:
-      releaseBranch: main
+      releaseBranch: np-fix-release-1
```
Comment (npepinpe) [inline .github/workflows/release-main-dry-run.yml:23]: :warning: I will revert this once approved before merging.

**Comment by npepinpe** [comment]
I still don't know why it only breaks during the release and not everywhere else. I have to imagine we set up the release stuff differently?

At any rate, this fixes it, and I don't see any downsides to having it there AFAIK.

**Comment by megglos** [comment]
I think https://github.com/camunda/zeebe/pull/14813 must have been the change which broke it, as the identity dependency was moved to other modules while the repo config still resided in the gateway pom. CC @remcowesterhoud @koevskinikola

**Review by megglos — APPROVED** [review]
🤞

**Comment by megglos** [comment]
bors merge

**Comment by megglos** [comment]
backport ping

**Pull request merged**